src/liballoc/arc.rs

```diff
@@ -282,6 +282,9 @@ impl<T> Arc<T> {
     /// To avoid a memory leak the pointer must be converted back to an `Arc` using
     /// [`Arc::from_raw`][from_raw].
     ///
+    /// If `T` is zero-sized (e.g. `Arc<()>`), the returned pointer address
+    /// will be meaningless.
```

Comment:
@eddyb can you elaborate on what you mean here? This method returns a pointer to the contents, right?

But it does seem weird to simultaneously say that the return value is *meaningless*, and then have `from_raw` say that the result must have been generated by `into_raw`.

I guess I'm not sure what *meaningless* means, really.